### ISSUE TITLE
[Feature Request]: Add p-values to Regression Coefficients table of Regularized Linear Regression 

### ISSUE THREAD

**Issue by kimsupit** [jasp-stats/jasp-issues#2106]
### Description

Can there please be p-values added to the Regression Coefficients table of Regularized Linear Regression 

### Purpose

To determine significane 

### Use-case

regularized regression 

### Is your feature request related to a problem?

No 

### Is your feature request related to a JASP module?

Machine Learning

### Describe the solution you would like

I would like an added column with p-values next to all regression coefficients 

### Describe alternatives that you have considered

None

**Comment by koenderks** [comment]
Hi!

I am sure where this p-value should coming from. Usually, GLMs and their penalized cousins provide model fit statistics, rather than p-values, to assess model fit. Do you have a reference on how to obtain such p-values and hoe to interpret them in this context?

If you want to do feature selection, why not just take all the coefficients that have not been shrunken to 0?

**Comment by kimsupit** [comment]
[image: Screenshot 2023-04-22 at 11.51.56.png]
Hi Koen,

I would like to know the p-values for all the x-variables to determine
their individual significance. I have a lot of variables with a coefficient
of which I can not determine for myself if it would be significant. My goal
is to identify (significant) predictors. In the table above you can see
that it displays the P-value. Would this be possible?


Looking forward to hearing from you.

Kind regards,

Kim


Op za 22 apr 2023 om 07:34 schreef Koen Derks ***@***.***>:

**Comment by koenderks** [comment]
If you are using regularized linear regression then some of the coefficients for the variables are shrunken to zero, which I guess you can interpret as a form of non-significance. Vice versa, all coefficients that are determined to be non-zero you could say are "significant" predictors.

On offering p-values in this output, the glmnet package that we use does not provide p-values for its coefficients and apparently there is also some discussion about whether you should obtain p-values for regularized regression coefficients. 

See https://stats.stackexchange.com/questions/410173/lasso-regression-p-values-and-coefficients

We'll keep an eye on this

**Comment by kimsupit** [comment]
Some of the variables have a coefficient of 0.004, 0.009, 0.010 and so on.
They are not zero but very small, would you say that these could still be
considered significant in this case? I know this depends on the context,
but you might have an idea. If you need more info, let me know :) Apologies
if these are kinda stupid questions.

Kind regards,

Kim

Op za 22 apr 2023 om 12:08 schreef Koen Derks ***@***.***>:

**Comment by koenderks** [comment]
No worries, it is a are very valid question that also comes up in this (https://stats.stackexchange.com/questions/478804/how-to-interpret-coefficients-from-regularized-cox-regression) discussion. 

In short, they emphasize that the coefficients are tailored to the training data and that one way to get an intuition about which predictors are important is to fit the model with different training sets (e.g., use a different seed in jasp) and see which predictors remain non-zero. The discussion refers to the book ISLR (https://www.statlearning.com, its free to download and read!), which contains some appropriate info about interpreting regularized regression coefficients. 

Hope this helps.

**Comment by tomtomme** [comment]
@kimsupit 
Closing, since this seems resolved. Pleas reopen, if I am mistaken!
